src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -255,60 +194,47 @@ public void clusterChanged(ClusterChangedEvent event) {
 
         // handle dangling indices, we handle those for all nodes that have a node file (data or master)
         if (nodeEnv.hasNodeFile()) {
-            if (danglingTimeout.millis() >= 0) {
-                synchronized (danglingMutex) {
-                    for (String danglingIndex : danglingIndices.keySet()) {
-                        if (newMetaData.hasIndex(danglingIndex)) {
-                            logger.debug("[{}] no longer dangling (created), removing", danglingIndex);
-                            DanglingIndex removed = danglingIndices.remove(danglingIndex);
-                            FutureUtils.cancel(removed.future);
-                        }
+            for (String danglingIndex : danglingIndices) {
+                if (newMetaData.hasIndex(danglingIndex)) {
+                    logger.debug("[{}] no longer dangling (created), removing", danglingIndex);
```

Comment:
I think while we're changing this, it would be nice to change this from "removing" to "removing from dangling index set", otherwise it is kind of confusing about whether data is actually being deleted (which it's not)